test/util/mock.go

```diff
@@ -0,0 +1,80 @@
+package util
```

Comment:
Why do we need an additional mockClient? Prefer to use the fake client provided by controller-runtime